### PR TITLE
tests: fix classic-ubuntu-core-transition race

### PR DESCRIPTION
**Pull request by fgimenez**
These changes set the ubuntu-core-transition-last-retry-time value under the data entry, instead of setting it at the top level of the state object, in order to properly stop the transition until the setup is done.

### CODE REVIEW

File: tests/main/classic-ubuntu-core-transition/task.yaml [hunk 1]
```diff
@@ -35,12 +32,11 @@ execute: |
...
     # installed
-    cat /var/lib/snapd/state.json | jq '.["ubuntu-core-transition-last-retry-time"]' > prevValue
-    systemctl stop snapd.service snapd.socket
-    cp /var/lib/snapd/state.json state.json.copy
-    now=$(date --utc --rfc-3339=ns | tr ' ' T)
-    cat state.json.copy | jq -c '. + {"ubuntu-core-transition-last-retry-time": "'"$now"'"}' > /var/lib/snapd/state.json
-    systemctl start snapd.service snapd.socket
+    systemctl stop snapd.{service,socket}
+    now=$(date --utc -Ins)
+    cat /var/lib/snapd/state.json | jq -c '. + {data: (.data + {"ubuntu-core-transition-last-retry-time": "'"$now"'"})}' > state.json.new
```
Comment (niemeyer) [inline tests/main/classic-ubuntu-core-transition/task.yaml]: Slightly simpler using this feature:
```
$ echo '{"a": {"b": 1}}' | jq '.a."c-d" = 2'
{
  "a": {
    "b": 1,
    "c-d": 2
  }
}
```

File: tests/main/classic-ubuntu-core-transition/task.yaml [hunk 2]
```diff
@@ -51,14 +47,10 @@ execute: |
...
-    systemctl stop snapd.service snapd.socket
-    cp /var/lib/snapd/state.json state.json.copy
-    if [ "$(cat prevValue)" = null ]; then
-        cat state.json.copy | jq -c 'del(.["ubuntu-core-transition-last-retry-time"])' > /var/lib/snapd/state.json
-    else
-        cat state.json.copy | jq -c '. + {"ubuntu-core-transition-last-retry-time": "'"$(cat prevValue)"'"}' > /var/lib/snapd/state.json
-    fi
-    systemctl start snapd.service snapd.socket
+    systemctl stop snapd.{service,socket}
+    cat /var/lib/snapd/state.json | jq -c 'del(.["data"]["ubuntu-core-transition-last-retry-time"])' > state.json.new
```
Comment (niemeyer) [inline tests/main/classic-ubuntu-core-transition/task.yaml]: Similarly:
```
$ echo '{"a": {"c-d": 1}}' | jq 'del(.a."c-d")'
{
  "a": {}
} 
```

File: tests/main/classic-ubuntu-core-transition/task.yaml [hunk 3]
```diff
@@ -7,9 +7,6 @@ systems: [-ubuntu-core-16-*, -ubuntu-*-ppc64el]
 warn-timeout: 1m
 kill-timeout: 5m
 
-restore: |
-    rm -f prevValue state.json.copy
```
Comment (pedronis) [inline tests/main/classic-ubuntu-core-transition/task.yaml]: you might want to keep  rm -f state.json.new to be on the safe side if the test breaks in the middle

Comment (fgimenez) [inline tests/main/classic-ubuntu-core-transition/task.yaml]: @pedronis done, thanks!

**Pull request merged**